testsuite/MDAnalysisTests/analysis/test_encore.py

```diff
@@ -140,30 +146,34 @@ def function(x):
 
         arguments = [tuple([i]) for i in np.arange(0,100)]
 
-        parallel_calculation = encore.utils.ParallelCalculation(function = function,
-                                                                ncores = 4,
-                                                                args = arguments)
+        parallel_calculation = encore.utils.ParallelCalculation(function=function,
+                                                                n_jobs=4,
+                                                                args=arguments)
         results = parallel_calculation.run()
 
         for i,r in enumerate(results):
             assert_equal(r[1], arguments[i][0]**2,
                 err_msg="Unexpeted results from ParallelCalculation")
 
-    def test_rmsd_matrix_with_superimposition(self):        
+    @dec.skipif(module_not_found('sklearn'),
+                "Test skipped because sklearn is not available.")
+    def test_rmsd_matrix_with_superimposition(self):
```

Comment:
I see. The only reason I'm dwelling on this issue is that I spent quite a lot of effort making sure that Encore could be run both with and without sklearn, supporting both options by creating front-ends for clustering and dimensionality that could use both our internal implementation and the sklearn implementations if they were available. 

Don't you think it is overkill to require sklearn (which is quite large, and can sometimes be difficult to compile), just because we require the smaller joblib library within? 

Personally, I would prefer adding a dependency explicitly to joblib - either as a MDAnalysis dependency or an Encore dependency. I'm assuming that your main argument against this is that it would be wasteful to have both dependencies for people who already have sklearn installed? But the same argument could be made in the opposite direction for users who do not need sklearn. I guess this all depends a bit on the fraction of MDAnalysis users we expect to have sklearn installed.